ja/lessons/advanced/otp-distribution.md

```diff
@@ -0,0 +1,486 @@
+---
+version: 1.0.0
+title: OTPディストリビューション
+---
+
+## ディストリビューションへの入門
+Elixirアプリは、単体のホストや複数のホストにまたがって分散された(distributed)一連の異なるノードで実行することができます。
+Elixirは、このレッスンで概説するいくつかの異なるメカニズムを通して、それらのノード間で通信をすることが可能です。
+
+{% include toc.html %}
+
+## ノード間通信
+
+ElixirはErlang VMで動作しますが、これはErlangの強力な [分散機能](http://erlang.org/doc/reference_manual/distributed.html) へアクセス可能であることを意味します。
+
+> 分散されたErlangシステムは互いに通信するいくつものErlangランタイムシステムを含んでいます。
+これらのそれぞれのランタイムはノードと呼ばれます。
+
+ノードは名前が与えられた任意のErlangランタイムシステムです。
+`iex` セッションを起動して名前を与えることで、ノードを開始できます:
+
+```bash
+iex --sname alex@localhost
+iex(alex@localhost)>
+```
+
+別のターミナルウィンドウで、もう一つのノードを起動してみましょう:
+
+```bash
+iex --sname kate@localhost
+iex(kate@localhost)>
+```
+
+これら2つのノードは `Node.spawn_link/1` を使うことでもう一方にメッセージを送ることができます。
+
+### `Node.spawn_link/1` による通信
+
+この関数は2つの引数を取ります:
+* 接続したいノードの名前
+* そのノードで動作しているリモートプロセスによって実行させる関数
+
+これはリモートノードとの間で通信を確立して、ノード上で与えられた関数を実行し、リンクされたプロセスのPIDを返します。
+
+`Kate` というモジュールを、その紹介方法を知っている `kate` ノードの中に定義しましょう:
+
+```elixir
+iex(kate@localhost)> defmodule Kate do
+...(kate@localhost)>   def say_name do
+...(kate@localhost)>     IO.puts "Hi, my name is Kate"
+...(kate@localhost)>   end
+...(kate@localhost)> end
+```
+
+#### メッセージの送信
+
+これで [`Node.spawn_link/2`](https://hexdocs.pm/elixir/Node.html#spawn_link/2) を使うと、 `alex` ノードは `kate` ノードに対して `say_name/0` 関数を呼ばせることができます:
+
+```elixir
+iex(alex@localhost)> Node.spawn_link(:kate@localhost, fn -> Kate.say_name end)
+Hi, my name is Kate
+#PID<10507.132.0>
+```
+
+#### 入出力とノードに関する注意
+
+注意が必要なのは、 `Kate.say_name/0` がリモートノードで実行されたとはいえ、 `IO.puts` の結果を受け取るのはローカル、つまり呼び出し側のノードだという点です。
+これは、ローカルノードが **グループリーダー** であるためです。
+Erlang VMはプロセスを通してI/Oを管理します。
+これによって `IO.puts` のようなI/Oタスクを分散されたノード間で実行することができます。
+これらの分散されたプロセスは、I/Oプロセスグループリーダーによって管理されます。
+このグループリーダーは常にプロセスを生産するノードとなります。
+そのため、 `alex` ノードは `spawn_link/2` から呼び出されているので、そのノードがグループリーダーとなり、 `IO.puts` の出力はそのノードの標準出力ストリームへと向けられます。
+
+#### メッセージへの返信
+
+受信したノードから送信側に対して何らかの *返信* を返したい場合はどうしましょう？ これは、単に `receive/1` と [`send/2`](https://hexdocs.pm/elixir/Process.html#send/3) を使うことで実現できます。
```

Comment:
Oh.. send/2 in caption and send/3 in a link.

Reply:
original text is wrong... https://github.com/elixirschool/elixirschool/pull/1841